Sources/Private/Model/Keyframes/KeyframeGroup.swift

```diff
@@ -111,10 +127,54 @@ final class KeyframeGroup<T>: Codable where T: Codable, T: Interpolatable {
       }
     }
   }
+}
 
-  // MARK: Private
+// MARK: Equatable
 
-  private enum KeyframeWrapperKey: String, CodingKey {
-    case keyframeData = "k"
+extension KeyframeGroup: Equatable where T: Equatable {
+  public static func == (_ lhs: KeyframeGroup<T>, _ rhs: KeyframeGroup<T>) -> Bool {
+    lhs.keyframes == rhs.keyframes
+  }
+}
+
+// MARK: Hashable
+
+extension KeyframeGroup: Hashable where T: Hashable {
+  public func hash(into hasher: inout Hasher) {
+    hasher.combine(keyframes)
+  }
+}
+
+// MARK: - Collection helpers
+
+extension KeyframeGroup {
+  // Maps the value of each individual keyframe in this group using the given closure
+  func map<Result>(_ mapping: (T) -> Result) -> KeyframeGroup<Result> {
+    KeyframeGroup<Result>(keyframes: .init(keyframes.map { keyframe in
+      keyframe.withValue(mapping(keyframe.value))
+    }))
+  }
+
+  // Maps the value of each individual keyframe in this group using the given closure,
+  // filtering out `nil`s.
+  func compactMap<Result>(_ mapping: (T) -> Result?) -> KeyframeGroup<Result> {
+    KeyframeGroup<Result>(keyframes: .init(keyframes.compactMap { keyframe in
+      guard let mappedValue = mapping(keyframe.value) else { return nil }
+      return keyframe.withValue(mappedValue)
+    }))
+  }
```

Comment:
nit: the closure passed to map is typically named `transform`
```suggestion
  // Maps the value of each individual keyframe in this group using the given closure,
  // filtering out `nil`s.
  func compactMap<Result>(_ transform: (T) -> Result?) -> KeyframeGroup<Result> {
    KeyframeGroup<Result>(keyframes: .init(keyframes.compactMap { keyframe in
      guard let mappedValue = transform(keyframe.value) else { return nil }
      return keyframe.withValue(mappedValue)
    }))
  }
```